api/envoy/http/common/base.proto

```diff
@@ -37,12 +37,16 @@ message Pattern {
   //     /dictionary/{term:1}/{term}
   //     /search{?q*,lang}
   //
-  string uri_template = 1;
+  string uri_template = 1 [(validate.rules).string = {
+    well_known_regex: HTTP_HEADER_VALUE,
```

Comment:
http_template may allow {}*, not sure if it is allowed in HTTP_HEADER_VALUE.

Reply:
`HTTP_HEADER_VALUE` with `strict = false` essentially allows all characters except for `\r\n\0`. This was specifically made to be compliant with Envoy, and Envoy uses this validation upstream as well. So `{}*` characters should be allowed.

Confirmed by:
- All integration tests pass
- https://github.com/envoyproxy/protoc-gen-validate/pull/323